gap/examples.gi

```diff
@@ -370,3 +370,889 @@ GeneralisedPetersenGraphCons);
+{n, k} -> StackedPrismGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(StackedPrismGraph, "for a function and two integers",
+[IsFunction, IsInt, IsInt], StackedPrismGraphCons);
+
+InstallMethod(WebGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D, i;
+  if (not IsInt(n)) or n < 3 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 2,");
+  fi;
+  D := StackedPrismGraph(IsMutableDigraph, n, 3);
+  for i in [1 .. (n - 1)] do
+    DigraphRemoveEdge(D, i, i + 1);
+    DigraphRemoveEdge(D, i + 1, i);
+  od;
+  return D;
+end);
+
+InstallMethod(WebGraphCons,
+"for IsImmutableDigraph, integer",
+[IsImmutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(WebGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(WebGraph, "for an integer", [IsInt],
+n -> WebGraphCons(IsImmutableDigraph, n));
+
+InstallMethod(WebGraph, "for a function and an integer",
+[IsFunction, IsInt], WebGraphCons);
+
+InstallMethod(HanoiGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D, C, i, nrVert, exp;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+  fi;
+  C := CycleGraph(IsMutableDigraph, 3);
+  D := C;
+  for i in [2 .. n] do
+    exp := 2 ^ (i - 2);
+    D := Digraph(IsMutableDigraph, []);
+    nrVert := DigraphNrVertices(C);
+    DigraphAddVertices(D, 3 * nrVert);
+    DigraphAddEdges(D, DigraphEdges(C));
+    DigraphAddEdges(D, DigraphEdges(C) + nrVert);
+    DigraphAddEdges(D, DigraphEdges(C) + (2 * nrVert));
+    DigraphAddEdge(D, nrVert / 2 + (1 / 2), nrVert + 1);
+    DigraphAddEdge(D, nrVert, 2 * nrVert + 1);
+    DigraphAddEdge(D, 2 * nrVert, DigraphNrVertices(D) - exp);
+    DigraphSymmetricClosure(D);
+
+    C := D;
+  od;
+  return D;
+end);
+
+InstallMethod(HanoiGraphCons,
+"for IsImmutableDigraph, integer",
+[IsImmutableDigraph, IsInt],
+function(filt, n)
+  local D;
+  D := MakeImmutable(HanoiGraphCons(IsMutableDigraph, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  return D;
+end);
+
+InstallMethod(HanoiGraph, "for an integer", [IsInt],
+n -> HanoiGraphCons(IsImmutableDigraph, n));
+
+InstallMethod(HanoiGraph, "for a function and an integer",
+[IsFunction, IsInt], HanoiGraphCons);
+
+InstallMethod(BinomialTreeGraphCons, "for IsMutableDigraph and an integer",
+[IsMutableDigraph, IsInt],
+function(filt, n)
+  local D, D_n, bits, i, is2n;
+  if (not IsInt(n)) or n < 1 then
+    ErrorNoReturn("the argument <n> must be an integer greater than 0,");
+  fi;
+  bits := Log(n, 2);
+  is2n := IsEvenInt(n) and IsPrimePowerInt(n);
+  if not is2n then
+    bits := bits + 1;
+  fi;
+  D := Digraph(IsMutableDigraph, []);
+  DigraphAddVertex(D, 1);
+  D_n := DigraphMutableCopy(D);
+  for i in [1 .. bits] do
```

Comment:
Is it possible to do this without repeatedly copying graphs? I haven't thought about it

Reply:
> Is it possible to do this without repeatedly copying graphs? I haven't thought about it

There is a definition with labelling vertices by binary numbers and having the subnodes of a node (as a tree) be obtained by changing a lower 0 to a 1, I need to do a test to see which is faster